### ISSUE TITLE
PF: mserver does not start with pathfinder option on Windows

### ISSUE THREAD

**Issue by monetdb-team** [MonetDB/MonetDB#2010]
Date: 2009-03-19 04:27:54 +0100
From: winston &lt;<wwinston>&gt;
To:   XQuery devs &lt;<bugs-xquery>&gt;
Version: -- development
CC:   alex_us01

Last updated: 2010-08-20 14:04:04 +0200

## Comment 11286
Date: 2009-03-19 04:27:54 +0100
From: winston &lt;<wwinston>&gt;

mserver appears to start OK.  However, the pathfinder option does not appear to start.  The start up process exits after the PF/Tijah module v0.9.0 is loaded; the start up exits to the DOS command prompt.  This is shown below under the heading: ACTUAL CONSOLE OUTPUT.   
 
 According the Tijah documentation, these following two steps should appear last and the command prompt should appear as shown here, and are thus missing:
  MonetDB/XQuery module v0.20.0 loaded
  XRPC administrative console at http://localhost:50001/admin  
 MonetDB>
 
 
 =x=x=x=x=x=
 ACTUAL CONSOLE OUTPUT:
 
 C:\MonetDB\MonetDB4>mserver
  MonetDB Server v4.28.0
  based on GDK   v1.28.0
  Copyright (c) 1993-July 2008, CWI. All rights reserved.
  Copyright (c) August 2008-, MonetDB B.V.. All rights reserved.
  Compiled for i686-pc-win32/32bit with 32bit OIDs; dynamically linked.
  Visit http://monetdb.cwi.nl/ for further information.
 MonetDB>Terminate batch job (Y/N)? y
 
 C:\MonetDB\MonetDB4>mserver --dbinit="module(pathfinder);"
  MonetDB Server v4.28.0
  based on GDK   v1.28.0
  Copyright (c) 1993-July 2008, CWI. All rights reserved.
  Copyright (c) August 2008-, MonetDB B.V.. All rights reserved.
  Compiled for i686-pc-win32/32bit with 32bit OIDs; dynamically linked.
  Visit http://monetdb.cwi.nl/ for further information.
  PF/Tijah module v0.9.0 loaded. http://dbappl.cs.utwente.nl/pftijah
 
 C:\MonetDB\MonetDB4>
 
 

## Comment 11287
Date: 2009-03-19 07:39:12 +0100
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

I am not sure, which Tijah documentation you refer to, but the MonetDB/XQuery documentation tells clearly how to start a MonetDB/XQuery server *on Windows*;
 http://monetdb.cwi.nl/XQuery/QuickTour/HelloWorld/index.html
 says
 "
 Starting Mserver
 
 The first task is to start the MonetDB Server. On Windows platforms this is done by clicking:
 
 Start -> Programs -> MonetDB -> MonetDB XQuery Server 
 
 If you are a Linux user, you should type something like:
 
   Mserver --dbinit="module(pathfinder);"
 "
 
 Alternatively, on Windows, you can also call "MXQserver.bat" to start the MonetDB/XQuery server.
 
 Maybe, you want to inform the maintainers of the Tijah documentation that you refer to to update their Windows-specific documentation accordingly.
 

## Comment 11288
Date: 2009-03-19 09:04:19 +0100
From: Henning Rode &lt;<hrode>&gt;

The PF/Tijah documentation was already pointing to MonetDB/XQuery pages here, but I added another direct link to the very helpful "first steps" page, Stefan suggested below.
 
 Does this solve the problem? Or do you still face problems to starting the Mserver with PF/Tijah?

## Comment 11289
Date: 2009-03-21 15:40:59 +0100
From: winston &lt;<wwinston>&gt;

I still have the same problem.  Your recommendation does not address the issue.  I have the same problem when executing the mxqserver command or from windows menu pick.  I merely showed you the command string so that you would see the problem in the MS-DOS command window.  
 Essentially, I can't get the XQuery part started.  The database startup exits without any error message and without giving me a command prompt.  Below, please observe the mxqserver output as contrasted to the mserver output; mserver starts OK and gives a MonetDB command prompt, but mxqserver does not start and does not give a MonetDB command prompt, instead exiting to a MS-DOS command prompt.
 
 C:\MonetDB\MonetDB4>mserver
  MonetDB Server v4.28.0
  based on GDK   v1.28.0
  Copyright (c) 1993-July 2008, CWI. All rights reserved.
  Copyright (c) August 2008-, MonetDB B.V.. All rights reserved.
  Compiled for i686-pc-win32/32bit with 32bit OIDs; dynamically linked.
  Visit http://monetdb.cwi.nl/ for further information.
 MonetDB>
 MonetDB>Terminate batch job (Y/N)? y
 
 C:\MonetDB\MonetDB4>mxqserver
  MonetDB Server v4.28.0
  based on GDK   v1.28.0
  Copyright (c) 1993-July 2008, CWI. All rights reserved.
  Copyright (c) August 2008-, MonetDB B.V.. All rights reserved.
  Compiled for i686-pc-win32/32bit with 32bit OIDs; dynamically linked.
  Visit http://monetdb.cwi.nl/ for further information.
  PF/Tijah module v0.9.0 loaded. http://dbappl.cs.utwente.nl/pftijah
 
 C:\MonetDB\MonetDB4>
 

## Comment 11290
Date: 2009-03-22 08:46:04 +0100
From: Henning Rode &lt;<hrode>&gt;

Can someone else take over here? This is so far not a PF/Tijah related problem. Since the XQuery server does not even start correctly, I am in fact not sure, if we should handle this as a bug. It seems to fall more in the category "installation problems". And if there is anyone else able to help you, we need to know also your system details: what kind of machine and windows version you are using.

## Comment 11291
Date: 2009-03-22 10:01:54 +0100
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Winston,
 
 here are a couple of questions for information that we'd need to analyse/debug your problem:
 
 - Which version of Windows are you using?
 
 - May we assume you installed http://monetdb.cwi.nl/downloads/Windows/Feb2009/MonetDB4-XQuery-Installer-i386-20090227.msi
   and no problems occured during installation?
 
 - Is this your first and only installation of MonetDB/XQuery, or did you upgrade/replace a previous version of MonetDB/XQuery?
   If the latter, which version of MonetDB/XQuery did you use before, and do you still have database created by that version on your machine?
 
 - Are the following directories writeable for you on your machine?
   %APPDATA%\MonetDB4
   %APPDATA%\MonetDB4\dbfarm
   %APPDATA%\MonetDB4\xquery_logs
 
 
 - Is general networking enabled or disabled on your machine?
 
 - Did you once manage to successfully start MonetDB with XQuery/pathfinder module, or did already the first attempt fail?
 
 - In case you once managed, what did you do/experience since between the last successful start-up and the first failing one?
   (Load & query documents? Encouter errors? If the latter, which ones?)
 
 - Do you happen to have any other servers/services running on your machine that use/block ports 50000 and/or 50001?
   You might try to start MonetDB/XQuery with alternative ports, e.g.,
   MXQserver.bat --mapi_port=50505
   MXQserver.bat --xrpc_port=55555
   MXQserver.bat --mapi_port=50505 --xrpc_port=55555
 
 
 Stefan
 

## Comment 11292
Date: 2009-03-22 22:27:56 +0100
From: winston &lt;<wwinston>&gt;

Windows XP Pro, SP3
 Yes, Installed MonetDB4-XQuery-Installer-i386-20090227.msi
 Replaced a previous version of MonetDB and I had troubles doing so: see Ticket 2687825 for more information.
 Yes, the directories are writable.  MonetDB did create the demo directory.
 Yes, networking is enabled.
 XQuery never started.  It failed on the first and all subsequent attempts.
 I tried switching the ports to 50505 and 55555, but I get the same results.
 
 Is there anyway to debug the start up process?
 
 
 
 

## Comment 11293
Date: 2009-03-23 09:18:43 +0100
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Thanks.
 
 In
 [ 2687825 ] Installation problem - MonetDB will not startup
 https://sourceforge.net/tracker/index.php?func=detail&aid=2687825&group_id=56967&atid=482468
 you say 
 "After installing a clean version of MoneyDB4 in a new directory, ..."
                                               ^^^^^^^^^^^^^^^^^^
 Are you sure that there are no remains of a previous installation of MonetDB left on your system?
 Did you next to %APPDATA%\MonetDB4\dbfarm also clean/remove %APPDATA%\MonetDB4\xquery_logs?
 
 Could you possibly consider uninstalling all versions of MonetDB from your system, remove all leftover remains of previous installation and their database (dbfarm) and logs (xquery_logs, sql_logs) and the re-install the latest installer from scratch?
 
 

## Comment 11294
Date: 2009-03-25 04:43:50 +0100
From: winston &lt;<wwinston>&gt;

I deleted the %APPDATA%/MonetDB directory and all files.
 
 I reran the MonetDB4-XQuery-Installer-i386-20090227.msi and selected remove MonetDB.
 
 I reran the MonetDB4-XQuery-Installer-i386-20090227.msi and reinstalled MonetDB.  
 
 The same exact problem is occuring.
 
 Is there any way to get more debug detail on the startup of MonetDB4?
 
 

## Comment 11295
Date: 2009-03-25 07:17:59 +0100
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Next to
 %APPDATA%\MonetDB
 did you also remove
 %APPDATA%\MonetDB4
                                 ^
 ?
 
 If not, please remove %APPDATA%\MonetDB4\dbfarm & %APPDATA%\MonetDB4\xquery_logs and try again --- otherwise, I'll try to think of some more debugging options and come back to you later ...
 

## Comment 11296
Date: 2009-03-25 13:18:00 +0100
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Winston,
 
 with removed or empty %APPDATA%\MonetDB4\dbfarm & %APPDATA%\MonetDB4\xquery_logs --- and in case starting MXQserver.bat still does not work,  could you please try to start a plain Mserver.bat, give the following commands in the given order on its "MonetDB>" console prompt, and check/report whether/which work without errors?
 
 module(mapi);
 module(pf_support);
 module(logger);
 module(mkey);
 module(xrpc_server);
 module(xrpc_client);
 module(pf_standoff);
 
 module(pathfinder);
 
 
 Stefan
 

## Comment 11297
Date: 2009-04-29 06:56:34 +0200
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Winston,
 
 is there any news regarding your problem?
 
 Did you test my latest suggestions?
 
 Stefan
 

## Comment 11298
Date: 2009-05-02 22:45:31 +0200
From: winston &lt;<wwinston>&gt;

Stefan
 
 This appears to have worked OK, but when I try to load documents from the client it fails:  
 
 xquery>pf:add-doc("C:\data\XMarkStandard\f0.xml","f0.xml","f0")
 more><>
 MAPI  = monetdb@localhost:50000
 QUERY = pf:add-doc("C:\data\XMarkStandard\f0.xml","f0.xml","f0")
 ERROR = !ERROR: interpret: no matching MIL operator to 'ws_create(int)'.
 xquery>
 
 

## Comment 11299
Date: 2009-05-02 23:31:10 +0200
From: winston &lt;<wwinston>&gt;

Stefan,
 
 I followed your procedure one more time, making sure to delete the %APPDATA%\MonetDB4\* files followed by the set of server commands that you gave me on 2009-03-25 08:18.
 
 This time I was able to get the server started and I was able to load documents through the client interface.
 
 I was able to load several documents except for one that was >1GB in size, so I opened a separate bug ticket for this problem.
 
 Thank you for your assistance.  You may close this bug.
 
   

## Comment 11300
Date: 2009-05-03 08:12:28 +0200
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Winston,
 
 thanks for the update!
 
 However, if only my work-around of loading all modules by hand works, but using MXQserver.bat or `Mserver.bat --dbinit="module(pathfinder);" does not, this bug report should stay open, and we should (try to) find out what is going wrong and (preferably) fit it (provided it is a problem inside MonetDB).
 
 Stefan
 

## Comment 11301
Date: 2009-05-05 23:49:03 +0200
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Same problem has also been reported in
 ID: 2787535 "XQuery Server doesn't start in Windows XP"
 https://sourceforge.net/tracker/?func=detail&atid=482468&aid=2787535&group_id=56967
 

## Comment 11302
Date: 2009-05-07 04:41:14 +0200
From: winston &lt;<wwinston>&gt;

The installation that I am having trouble with is a Dell laptop with Windows XP SP3, Intel Core 2.
 
 ACTION TAKEN: I installed MonetDB on one of my older HP PCs with Win XP SP2, Intel Pentium 4.  
 
 RESULTS: The WinXP SP2 installation seems to be working OK.  The problems I am encountering are all on the WinXP SP3 on the Dell.

## Comment 11303
Date: 2009-05-07 06:49:38 +0200
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Winston,
 
 thanks for the update!
 Do I understand correctly, that on the WinXP SP2 starting the MonetDB/XQuery server via the start menu entry and/or via MXQserver.bat works fine, while it fails on the WinXP SP3 machine?
 The hardware should not make a difference, so I expect the problems to be related to WinXP SP2 vs. SP3.
 Are both of your Windows 32-bit?
 
 
 
 Alex (who posted ID: 2787535),
 
 which version of Windows XP are you using?
 SP2 or SP3? 
 32-bit or 64-bit?
 
 
 
 Stefan
 

## Comment 11304
Date: 2009-05-07 15:21:28 +0200
From: Sjoerd Mullender &lt;<sjoerd>&gt;

Winston,
 
 I have a couple of questions for you:
 
 - which older version of MonetDB did you replace with the new installation?
 - the default location of older versions of MonetDB was C:\Program Files\CWI\MonetDB4, whereas now it is C:\Program Files\MonetDB\MonetDB4.  In which location did you install, and are there any files left over in the old location?
 - still older versions of MonetDB installed a few files in C:\Windows\system32 (files matching the pattern lib*.dll).  We don't do that anymore.  Do you have any of those files still lying around?
 - the default database location is %APPDATA%\MonetDB4.  Did you remove that folder?
 - older versions of MonetDB used %APPDATA%\MonetDB (note the missing 4) as the location of the database.  The Mserver.bat script has code to use that folder if the MonetDB4 folder does not exist.  Did you have the MonetDB folder?
 
 The installers are built on a Windows XP Professional SP3 system and have also been tested on such a system.  I was able to start the server from the Start menu and do a simple query.

## Comment 11305
Date: 2009-05-08 04:03:44 +0200
From: winston &lt;<wwinston>&gt;

The version of windows XP is 32-bit.  I have several software products installed, including DB2 and SQL Server.
 
 I don't recall the older MonetDB versions I replaced, but there were a couple going back a few years.  Previously, I also installed and removed MonetDB5.  I have no C:\Program Files\CWI directory.
 
 I installed the March 2009 3/24 version of MonetDB.  I installed this version in C:\MonetDB\MonetDB4 directory.  Today, I removed and reinstalled MonetDB in the default location Program Files.  
 
 I found libfdnvin.dll in C:\Windows\system32 with a date of 6/8/2000.  I removed this file from this directory.  Also removed a file call libeay32.dll.  (OpenSSL ?)
 
 The same behavior occurred.
 
 Is there a way to trace what files MonetDB is loading from what directories?
 
 
 
 
 

## Comment 11306
Date: 2009-05-08 10:15:20 +0200
From: Sjoerd Mullender &lt;<sjoerd>&gt;

To see which files are loaded by MonetDB, you can do the following:
 
 Go to the Microsoft Sysinternals website (http://technet.microsoft.com/en-us/sysinternals/default.aspx).  On the left of the page there is a link to Sysinternals Suite.  Go there.  Then either download the whole suite (top right of the page), or just the Process Monitor (which is part of the suite).  Install the download (i.e. extract somewhere) and start the Process Monitor (Procmon.exe).
 
 After accepting the license, select the menu entry Filter -> Filter... (or click on the Filter toolbutton).  In the pop up, change the fields at the top to read:
 Process Name     is    Mserver.exe    then   Include
 then Add and Apply (or OK).
 
 Then start MonetDB XQuery Server from the Start menu.
 
 After this you can File -> Save... the captured events.  Save them as CSV and attach the file to this bug report.

## Comment 11307
Date: 2009-05-09 17:39:16 +0200
From: winston &lt;<wwinston>&gt;

Trace added.  Split into two files due to size restrictions

## Comment 11308
Date: 2009-05-12 17:59:29 +0200
From: Sjoerd Mullender &lt;<sjoerd>&gt;

I see lots of differences between your log and one I made on my system.
 
 Can you try removing the file C:\WINDOWS\Prefetch\MSERVER.EXE-2DF2C547.pf and then try again?
 This file has to do with prefetching.  It should make loading the program faster.  I can imagine it somehow got corrupted, so if you remove it, perhaps it works better.
 
 Lines 8 to 14973 in your log are equivalent to lines 8 to 11 in mine.  So there is a *huge* difference.

## Comment 11309
Date: 2009-05-19 04:16:25 +0200
From: winston &lt;<wwinston>&gt;

Why the difference?  In the DOS window, before starting MServer, I cleared out the path environment variable.  The size was reduced, but it is still many hundreds of lines.
 
 What should the environment variables look like?  
 
 I tried to delete the file: C:\WINDOWS\Prefetch\MSERVER.EXE-2DF2C547.pf, but it is no longer there.
 
 No difference after emptying the path variable.

## Comment 11310
Date: 2009-05-20 16:10:24 +0200
From: Sjoerd Mullender &lt;<sjoerd>&gt;

I just attached two files to this bug.  One contains a copy/paste of a cmd session of a successful start with the output of the "set" command to show my environment.  It's pretty small.  The other file is a zip containing two log dump from procmon for two different successful runs.  Notice that both these logs are significantly smaller than yours (two of them together fit easily within the sourceforge limit).
 
 I compared your log with mine by removing the first few columns from the logs (up to and including the PID column) and then using diff.
 
 In the mean time, I have no idea what the problem can be.

## Comment 11311
Date: 2009-06-04 22:28:39 +0200
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Just for the records, here's a small correction for my suggested work-around of 2009-03-25 13:18 :
 
 If loading modules individually, this should happen in the following order:
 
 module(pf_support);
 module(logger);
 module(mkey);
 module(xrpc_server);
 module(xrpc_client);
 module(pf_standoff);
 module(pathfinder);
 module(mapi);
 
 i.e., module "mapi" should be last, not first.
 
 Stefan
 

## Comment 11312
Date: 2009-10-11 20:29:53 +0200
From: winston &lt;<wwinston>&gt;

Reinstalled Windows XP from scratch -> installed MonetDB -> everything works OK.

## Comment 11313
Date: 2010-05-04 09:32:09 +0200
From: Pseudo user for Sourceforge import &lt;<sfimport>&gt;

This bug was previously known as tracker item 2693363 at http://sourceforge.net/support/tracker.php?aid=2693363
 
 The original assignee of this bug does not have
    an account here. Reassigning to the default assignee
    for the component, bugs-xquery@monetdb.org.
    Previous assignee was nobody@users.sourceforge.net.
 
 

## Comment 14116
Date: 2010-06-17 15:44:49 +0200
From: Sjoerd Mullender &lt;<sjoerd>&gt;

*** Bug #2068 has been marked as a duplicate of this bug. ***

## Comment 15809
Date: 2011-05-16 21:50:58 +0200
From: Niels Nes &lt;<niels>&gt;

Changeset [f169139660d8](https://dev.monetdb.org/hg/MonetDB?cmd=changeset;node=f169139660d8)  made by Niels Nes <niels@cwi.nl> in the MonetDB repo, refers to this bug.
 
 For complete details, see [http//devmonetdborg/hg/MonetDB?cmd=changeset;node=f169139660d8](https://dev.monetdb.org/hg/MonetDB?cmd=changeset;node=http//devmonetdborg/hg/MonetDB?cmd=changeset;node=f169139660d8) 
 
 Changeset description:
 
 	add test for count-count-distinct bug #2808
 
 	check variable sized types on binary copy into (solves bug #2010).
